Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -590,6 +594,32 @@ theorem pdiv_zeta [DivisionSemiring R] (f : ArithmeticFunction R) :
 
 end Pdiv
 
+section ProdToFinsetFactors
+
+/-- The map $n \mapsto \prod_{p \mid n} f(p)$ as an arithmetic function -/
+def prodToFinsetFactors [CommMonoidWithZero R] (f : ℕ → R) : ArithmeticFunction R where
+  toFun d := if d = 0 then 0 else ∏ p in d.factors.toFinset, f p
+  map_zero' := if_pos rfl
+
+open Std.ExtendedBinder
+
+/-- `∏ᵖ p ∣ n, f p` is custom notation for `prodToFinsetFactors f n` -/
+scoped syntax (name := bigproddvd) "∏ᵖ " extBinder " ∣ " term ", " term:67 : term
+scoped macro_rules (kind := bigproddvd)
+  | `(∏ᵖ $x:ident ∣ $n, $r) => `(prodToFinsetFactors (fun $x ↦ $r) $n)
+
+@[simp]
+theorem prodToFinsetFactors_apply [CommMonoidWithZero R] {f: ℕ → R} {n : ℕ} [hn : NeZero n] :
+    ∏ᵖ p ∣ n, f p = ∏ p in n.factors.toFinset, f p :=
+  if_neg (hn.ne)
+
+theorem prodToFinsetFactors_apply_of_ne_zero [CommMonoidWithZero R] {f: ℕ → R} {n : ℕ}
```

Comment:
```suggestion
theorem prodToFinsetFactors_apply_of_ne_zero [CommMonoidWithZero R] { f: ℕ → R} {n : ℕ}
```

Reply:
See [zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/toFinsetFactors)

Reply:
This was resolved by #8164